editoast/src/models/pathfinding.rs

```diff
@@ -96,3 +92,189 @@ pub struct PathWaypoint {
     pub position: f64,
     pub suggestion: bool,
 }
+
+impl Pathfinding {
+    pub fn track_section_ids(&self) -> HashSet<String> {
+        self.payload
+            .route_paths
+            .iter()
+            .flat_map(|route_path| &route_path.track_ranges)
+            .map(|track_range| track_range.track.to_string())
+            .collect::<HashSet<_>>()
+    }
+
+    pub fn merged_track_ranges(&self) -> Vec<DirectionalTrackRange> {
+        let mut res = Vec::new();
+        let mut track_ranges = self
+            .payload
+            .route_paths
+            .iter()
+            .flat_map(|route_path| &route_path.track_ranges)
+            .filter(|track_range| track_range.begin != track_range.end);
+        let mut current_range = track_ranges.next().unwrap().clone();
+        for track_range in track_ranges {
+            assert!(track_range.begin < track_range.end);
+            if track_range.track == current_range.track {
+                assert!(track_range.direction == current_range.direction);
+                match current_range.direction {
+                    Direction::StartToStop => {
+                        assert!(current_range.end == track_range.begin);
+                        current_range.end = track_range.end;
+                    }
+                    Direction::StopToStart => {
+                        assert!(current_range.begin == track_range.end);
+                        current_range.begin = track_range.begin;
+                    }
+                }
+            } else {
+                res.push(current_range.clone());
+                current_range = track_range.clone();
+            }
+        }
+        res.push(current_range);
+        res
+    }
+}
+
+impl Identifiable for Pathfinding {
+    fn get_id(&self) -> i64 {
+        self.id.unwrap()
+    }
+}
+
+#[cfg(test)]
+pub mod tests {
+    use super::*;
+
+    pub fn simple_pathfinding(infra_id: i64) -> Pathfinding {
```

Comment:
That's nice! But why not turn it into a fixture?

Reply:
You can do this but not for a given infra. 

Two options:

- We can have a fixture that creates both an infra and a path.
- We can ignore the infra_id parameter and set it to `0` since it's not used in tests

Reply:
It's not used in tests but it is necessary that it is a valid infra_id to write it in the db

I wonder if we even need to return a `TestFixture` in this case, since the delete of the infra should cascade the deletion of the objects referencing it